common/src/main/java/feast/common/logging/AuditLogger.java

```diff
@@ -113,28 +130,52 @@ private static void log(Level level, AuditLogEntry entry) {
       return;
     }
     if (entry.getKind().equals(AuditLogEntryKind.MESSAGE)
-        && !properties.isMessageLoggingEnabled()) {
+        && !properties.getMessageLogging().isEnabled()) {
       return;
     }
 
-    // Log event to audit log through enabled formats
-    String entryJSON = entry.toJSON();
-    switch (level) {
-      case TRACE:
-        log.trace(AUDIT_MARKER, entryJSON);
-        break;
-      case DEBUG:
-        log.debug(AUDIT_MARKER, entryJSON);
-        break;
-      case INFO:
-        log.info(AUDIT_MARKER, entryJSON);
-        break;
-      case WARN:
-        log.warn(AUDIT_MARKER, entryJSON);
-        break;
-      case ERROR:
-        log.error(AUDIT_MARKER, entryJSON);
-        break;
+    // Either forward log to logging layer or log to console
+    if (properties.getMessageLogging().getDestination().equals(DESTINATION)) {
+      Map<String, Object> fluentdLogs = new HashMap<>();
+      String releaseName =
+          StringUtils.defaultIfEmpty(System.getenv("RELEASE_NAME"), DEFAULT_RELEASE_NAME);
+      MessageAuditLogEntry messageAuditLogEntry = (MessageAuditLogEntry) entry;
+
+      fluentdLogs.put("id", messageAuditLogEntry.getId());
+      fluentdLogs.put("service", messageAuditLogEntry.getService());
+      fluentdLogs.put("status_code", messageAuditLogEntry.getStatusCode());
+      fluentdLogs.put("method", messageAuditLogEntry.getMethod());
+      fluentdLogs.put("release_name", releaseName);
+      try {
+        fluentdLogs.put("request", JsonFormat.printer().print(messageAuditLogEntry.getRequest()));
+        fluentdLogs.put("response", JsonFormat.printer().print(messageAuditLogEntry.getResponse()));
+      } catch (InvalidProtocolBufferException e) {
+        throw io.grpc.Status.NOT_FOUND
```

Comment:
NOT_FOUND and description are not really represent the cause of the exception that may happen here

Reply:
Updated description and status type.